app/assets/javascripts/backbone/views/timeline_table_view.coffee

```diff
@@ -71,8 +76,9 @@ class MS.Views.TimelineTableView extends Backbone.View
     @stepModal.new(@projectId, @refresh.bind(@))
 
   showStep: (e) ->
-    e.preventDefault()
-    @stepModal.show(@stepIdFromEvent(e), @refresh.bind(@))
+    if @$(e.target).hasClass('project-step')
```

Comment:
Maybe instead of asking if this is the `td` that we are clicking on, we can check to see if it's the checkbox, the more link, or the icon. If any of these, we should not open the modal. If not these, then we should.

This may address one of QA issues I raised earlier.